### PR TITLE
Attekmi: not required partnerName hotfix

### PR DESCRIPTION
**Pull request by SmartHubSolutions**
This hotfix is ​​required for the following PRs:
https://github.com/prebid/prebid.github.io/pull/5836#discussion_r1952981841
https://github.com/prebid/prebid.github.io/pull/5822#pullrequestreview-2612781433

### CODE REVIEW

**Comment by SmartHubSolutions** [comment]
@SyntaxNode @bsardo Please review it.

File: adapters/smarthub/params_test.go [hunk 1]
```diff
@@ -35,11 +35,9 @@ var invalidParams = []string{
 	`{"partnerName":"partnertest"}`,
 	`{"seat":"9Q20EdGxzgWdfPYShScl"}`,
 	`{"token":"Y9Evrh40ejsrCR4EtidUt1cSxhJsz8X1"}`,
-	`{"seat":"9Q20EdGxzgWdfPYShScl", "token":"alNYtemWggraDVbhJrsOs9pXc3Eld32E"}`,
```
Comment (bsardo) [inline adapters/smarthub/params_test.go:38]: Since none of your params are now considered required, please add the following test cases to `validParams` to ensure that the absence of each param is valid, the empty string is allowed for `partnerId` and a minimum length of 1 is valid for `seat` and `token`:
```
{"partnerName":""}
{"partnerName":"1"}
{"seat":"1"},
{"token":"1"},
```

File: adapters/smarthub/smarthub.go [hunk 2]
```diff
@@ -60,10 +60,14 @@ func (a *adapter) getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ExtSmartHub,
...
 	endpointParams := macros.EndpointTemplateParams{
-		Host:      params.PartnerName,
 		AccountID: params.Seat,
 		SourceId:  params.Token,
 	}
+
+	if params.PartnerName != "" {
+		endpointParams.Host = params.PartnerName
+	}
+
```
Comment (bsardo) [inline adapters/smarthub/smarthub.go:70]: You shouldn't need to do this. You can just keep the original logic so that `Host` is always assigned the value of `params.PartnerName`. If `params.PartnerName` is an empty string, the generated URL will contain `?partnerName=&`.

Were you trying to conditionally exclude the `partnerName` query param?

I suggest either modifying an existing JSON test or adding a new JSON test where the partner name is either an empty string or not specified to verify your intended behavior.

File: adapters/smarthub/params_test.go (hunk 1 above)
Comment (SmartHubSolutions) [reply adapters/smarthub/params_test.go:38]: ok I added this test cases and all tests was passed

File: adapters/smarthub/smarthub.go (hunk 2 above)
Comment (SmartHubSolutions) [reply adapters/smarthub/smarthub.go]: Yes, you're right, in our system when partner use alias-adapter we not reading parnerName param, doesn't matter whether there is data or not. I returned first variant.

File: adapters/smarthub/smarthub.go [hunk 3]
```diff
@@ -60,10 +60,11 @@ func (a *adapter) getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ExtSmartHub,
 
 func (a *adapter) buildEndpointURL(params *openrtb_ext.ExtSmartHub) (string, error) {
 	endpointParams := macros.EndpointTemplateParams{
-		Host:      params.PartnerName,
 		AccountID: params.Seat,
 		SourceId:  params.Token,
+		Host:      params.PartnerName,
```
Comment (SyntaxNode) [inline adapters/smarthub/smarthub.go:65]: Might be better to delete the Host template param entirely. This is only used for endpoint macros and neither smarthub nor its aliases use this macro.

File: adapters/smarthub/params_test.go (hunk 1 above)
Comment (SyntaxNode) [reply adapters/smarthub/params_test.go:38]: > Since none of your params are now considered required

@bsardo  - The seat and token params are still required in this schema.

File: adapters/smarthub/smarthub.go (hunk 3 above)
Comment (SmartHubSolutions) [reply adapters/smarthub/smarthub.go:65]: Currently, we use both methods of redirecting traffic, both through partnerName and using aliases.We also need to use param partnerName to temporarily connect partners.

File: adapters/smarthub/params_test.go (hunk 1 above)
Comment (SmartHubSolutions) [reply adapters/smarthub/params_test.go:38]: Hmm this cases I added to notValidParams because we using  parnterName+seat+token OR seat+token in aliases, this means that the pair seat+token will always be required.

File: adapters/smarthub/smarthub.go (hunk 3 above)
Comment (SyntaxNode) [reply adapters/smarthub/smarthub.go:65]: Ah, I see. The aliases do not use this macro, but the base `smarthub` adapter does. Ok.

**Review by SyntaxNode — DISMISSED** [review]

File: adapters/smarthub/params_test.go (hunk 1 above)
Comment (bsardo) [reply adapters/smarthub/params_test.go:38]: > > Since none of your params are now considered required
> 
> @bsardo - The seat and token params are still required in this schema.

Oops not sure what I was looking at when I reviewed this 😄

Comment (bsardo) [reply adapters/smarthub/params_test.go:38]: @SmartHubSolutions can you please add the following tests to `validParams` to verify what is required and minimum lengths:
```
`{"partnerName":"", "seat":"1", "token":"1"}`,
`{"seat":"1", "token":"1"}`,
```

Comment (SmartHubSolutions) [reply adapters/smarthub/params_test.go:38]: no problem I addeed them, tests passed

**Review by bsardo — APPROVED** [review]

**Review by SyntaxNode — APPROVED** [review]
